Sources/Commands/PackageCommands/APIDiff.swift

```diff
@@ -173,39 +194,180 @@ struct APIDiff: AsyncSwiftCommand {
+            baselinesDirectory: baselineDir,
+            modulesToCompare: modulesWithBaselines,
+            breakageAllowListPath: breakageAllowlistPath
+        )
+        let delegate = DiagnosticsCapturingBuildSystemDelegate()
+        let buildSystem = try await swiftCommandState.createBuildSystem(
+            traitConfiguration: .init(traitOptions: self.traits),
+            cacheBuildManifest: false,
+            productsBuildParameters: productsBuildParameters,
+            delegate: delegate
+        )
+        try await buildSystem.build()
+
+        // Report the results of the comparison.
+        var comparisonResults: [SwiftAPIDigester.ComparisonResult] = []
+        for (targetName, diagnosticPaths) in delegate.serializedDiagnosticsPathsByTarget {
+            guard let targetName, !diagnosticPaths.isEmpty else {
+                continue
+            }
+            var apiBreakingChanges: [SerializedDiagnostics.Diagnostic] = []
+            var otherDiagnostics: [SerializedDiagnostics.Diagnostic] = []
+            for path in diagnosticPaths {
+                let contents = try swiftCommandState.fileSystem.readFileContents(path)
+                guard contents.count > 0 else {
+                    continue
+                }
+                let serializedDiagnostics = try SerializedDiagnostics(bytes: contents)
+                let apiDigesterCategory = "api-digester-breaking-change"
+                apiBreakingChanges.append(contentsOf: serializedDiagnostics.diagnostics.filter { $0.category == apiDigesterCategory })
+                otherDiagnostics.append(contentsOf: serializedDiagnostics.diagnostics.filter { $0.category != apiDigesterCategory })
+            }
+            let result = SwiftAPIDigester.ComparisonResult(
+                moduleName: targetName,
+                apiBreakingChanges: apiBreakingChanges,
+                otherDiagnostics: otherDiagnostics
+            )
+            comparisonResults.append(result)
+        }
+
+        var detectedBreakingChange = false
+        for result in comparisonResults.sorted(by: { $0.moduleName < $1.moduleName }) {
+            if result.hasNoAPIBreakingChanges && !modulesToDiff.contains(result.moduleName) {
+                continue
+            }
+            try printComparisonResult(result, observabilityScope: swiftCommandState.observabilityScope)
+            detectedBreakingChange = detectedBreakingChange || !result.hasNoAPIBreakingChanges
+        }
+
+        for module in modulesToDiff.subtracting(modulesWithBaselines) {
+            print("\nSkipping \(module) because it does not exist in the baseline")
+        }
+
+        if detectedBreakingChange {
+            throw ExitCode(1)
+        }
+    }
+
+    private func generateAPIBaselineUsingIntegratedAPIDigesterSupport(_ swiftCommandState: SwiftCommandState, baselineRevision: Revision, baselineDir: Basics.AbsolutePath, modulesNeedingBaselines: Set<String>) async throws -> Set<String> {
```

Comment:
suggestion: Consider created automated tests that validate this function in isolation with happy path, and also fault injection, and more.